capstone/src/main/java/com/google/sps/servlets/GroupPostDataServlet.java

```diff
@@ -73,21 +97,50 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     // Creates entity with submitted data and add to database
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(createPostEntity(timestamp, authorId, postText, challengeName, img, likes, comments));
+    datastore.put(createPostEntity(timestamp, authorName, postText, challengeName, img, likes, comments));
 
     // Redirect back to the HTML page.
     response.sendRedirect("/group.html");
   }
 
-  public Entity createPostEntity(long timestamp, String authorId, String postText, String challengeName, String img, ArrayList<String> likes, ArrayList<Comment> comments) {
+  private Entity createPostEntity(long timestamp, String authorName, 
+    String postText, String challengeName, String img, ArrayList<String> likes, ArrayList<Comment> comments) {
     Entity taskEntity = new Entity("Post");
-    taskEntity.setProperty("authorId", authorId);
+    taskEntity.setProperty("authorId", authorName);
     taskEntity.setProperty("timestamp", timestamp);
     taskEntity.setProperty("postText", postText);
     taskEntity.setProperty("challengeName", challengeName);
     taskEntity.setProperty("img", img);
     taskEntity.setProperty("likes", likes);
-    taskEntity.setProperty("comments", comments);
+    taskEntity.setProperty("comments", createCommentEntities(comments));
     return taskEntity;
 	}
+
+  // Create list of comment entities from list of comment objects
+  private ArrayList<EmbeddedEntity> createCommentEntities(ArrayList<Comment> comments) {
+    ArrayList<EmbeddedEntity> allComments = new ArrayList<>();
+    for (Comment comment: comments) {
+      EmbeddedEntity commentEntity = new EmbeddedEntity();
+      commentEntity.setProperty("timestamp", comment.getTimestamp());
+      commentEntity.setProperty("commentText", comment.getCommentText());
+      commentEntity.setProperty("userId", comment.getUser());
+      allComments.add(commentEntity);
```

Comment:
I think long-term I'd like to see all entity work inside the Objects directory instead of in the servlets

Reply:
sure! I'll make those changes in another PR